enterprise/meta.go

```diff
@@ -404,6 +409,22 @@ func (d *defaultClient) Do(URL *url.URL, path, method string, params map[string]
 
 }
 
+// AuthedCheckRedirect tries to follow the Influx Enterprise pattern of
+// redirecting to the leader but preserving authentication headers.
+func AuthedCheckRedirect(req *http.Request, via []*http.Request) error {
+	if len(via) >= 10 {
+		return errors.New("too many redirects")
+	} else if len(via) == 0 {
+		return nil
+	}
+	for attr, val := range via[0].Header {
```

Comment:
Good point @timraymond... the headers stripped out are here: https://golang.org/src/net/http/client.go#L823

So, yup, Ill just add in `Authorization`